### PR TITLE
Update search contexts documentation

### PR DESCRIPTION
**Pull request by christinaforney**

### CODE REVIEW

File: doc/code_search/explanations/features.md [hunk 1]
```diff
@@ -125,19 +125,19 @@ Indexing multiple branches will add additional resource requirements to Sourcegr
...
 
 Sourcegraph Cloud currently supports two search contexts: 
 
 - Your personal context, `context:@username`, which automatically includes all repositories you add to the Sourcegraph instance.
 - The global context, `context:global`, which includes all repositories on the Sourcegraph instance.
 
-Search contexts are currently only available on Sourcegraph Cloud.
+Self-hosted Sourcegraph instances currently have this functionality disabled, because it is under active development. To enable on your instance use the following config in your user or organization settings.
+
+> NOTE: This will allow any user at your organization to add code from GitHub.com and GitLab.com to your instance using their personal token.
```
Comment (novoselrok) [inline doc/code_search/explanations/features.md:137]: The `showSearchContext` flag only enables the search context dropdown selector, it does not automatically allow adding public code. That is controlled by a separate flag `externalService.userMode` (@ryanslade please confirm).

Comment (christinaforney) [reply doc/code_search/explanations/features.md:137]: Oh interesting. In that case, what would the experience be for a user that only enabled contexts but didn't have the ability to add code?

Comment (novoselrok) [reply doc/code_search/explanations/features.md:137]: Right now, nothing would happen (user has to add at least one repo). Once we add context management in next few weeks they will be able to create arbitrary contexts themselves from all available repos.

Comment (ryanslade) [reply doc/code_search/explanations/features.md:137]: @novoselrok Correct, they need to set this in site config in order to enable user added public code:

```json
"externalService.userMode": "public",
```

Comment (christinaforney) [reply doc/code_search/explanations/features.md:137]: Ok thanks, I'll make a change to this section to fix this!

Comment (christinaforney) [inline doc/code_search/explanations/features.md:137]:
```suggestion
This feature is currently under active development for self-hosted Sourcegraph instances and is therefore disabled by default. If enabled on your instance with the below configuration, users will not see the search context functionality until at least one repository is added by that user (which is not yet available on private instances). 

To enable search contexts on your private instance, use the following config in your user or organization settings:
```

Comment (novoselrok) [reply doc/code_search/explanations/features.md:137]: Yep, that looks good 👍

Comment (ryanslade) [reply doc/code_search/explanations/features.md:137]: Look good!

**Review by novoselrok — APPROVED** [review]

**Pull request merged**